src/components/RadioGroup/RadioGroup.tsx

```diff
@@ -1,114 +1,148 @@
+  /** The `name` prop indicates the form group for all the Radio children. */
+  name: string;
   /** The action to perform on the `<input>`'s onChange function  */
-  onChange?: (value: string) => string;
-  /** Offers the ability to show the radio's label onscreen or hide it. Refer to the `labelText` property for more information. */
-  showLabel: boolean;
-  /** Populates the value of the input */
-  defaultValue?: string;
-  layout?: RadioGroupLayouts;
+  onChange?: (value: string) => void;
+  /** Whether or not to display "Required"/"Optional" in the label text. */
+  optReqFlag?: boolean;
+  /** Offers the ability to show the group's legend onscreen or hide it. Refer
+   * to the `labelText` property for more information. */
+  showLabel?: boolean;
 }
 
 export const onChangeDefault = () => {
   return;
 };
 
-const directionMap: DirectionMap = {
-  [RadioGroupLayouts.Vertical]: "column",
-  [RadioGroupLayouts.Horizontal]: "row",
-};
+const RadioGroup = React.forwardRef<HTMLInputElement, RadioGroupProps>(
+  (props, ref?) => {
+    const {
+      children,
+      className = "",
+      defaultValue,
+      errorText,
+      helperText,
+      id = generateUUID(),
+      isDisabled = false,
+      isInvalid = false,
+      isRequired = false,
+      labelText,
+      layout = RadioGroupLayoutTypes.Column,
+      name,
+      onChange = onChangeDefault,
+      optReqFlag = true,
+      showLabel = true,
+    } = props;
+    const footnote = isInvalid ? errorText : helperText;
+    const direction = layout;
+    const newChildren = [];
 
-function RadioGroup(props: React.PropsWithChildren<RadioGroupProps>) {
-  const {
-    className,
-    id = generateUUID(),
-    children,
-    name,
-    defaultValue,
-    onChange = onChangeDefault,
-    layout = RadioGroupLayouts.Vertical,
-    labelText,
-    helperText,
-    errorText,
-    showLabel,
-    errored = false,
-    required,
-    disabled,
-  } = props;
-  const baseName = "RadioGroup";
-  const footnote = errored ? errorText : helperText;
-  const direction = directionMap[layout];
-  const newChildren = [];
+    // Use Chakra's RadioGroup hook to set and get the proper props
+    // or the custom components.
+    const { getRootProps, getRadioProps } = useRadioGroup({
+      name,
+      defaultValue,
+      onChange,
+    });
+    const radioGroupProps = getRootProps();
 
-  React.Children.map(children, (child: React.ReactElement, i) => {
-    const attributes = { key: i };
-    if (child !== undefined && child !== null) {
-      attributes["name"] = name;
-      if (disabled) {
-        attributes["disabled"] = true;
-        attributes["aria-disabled"] = true;
-      }
-      if (errored) {
-        attributes["errored"] = true;
-      }
-      if (child.props.value === defaultValue) {
-        attributes["checked"] = true;
+    // Go through the Radio children and update them as needed.
+    React.Children.map(children, (child: React.ReactElement, i) => {
```

Comment:
Can you add validation here to ensure that only DS `Radio` components are being passed?